src/main/java/com/salesforce/BenchmarkApp.java

```diff
@@ -0,0 +1,315 @@
+        kafkaProducerConfig.put(ProducerConfig.ACKS_CONFIG, kafkaAcks);
+        kafkaProducerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, IntegerSerializer.class.getName());
+        kafkaProducerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, ByteArraySerializer.class.getName());
+
+        // Global counters
+        Counter topicsCreated = Counter
+                .builder("numTopicsCreated")
+                .description("Number of topics we've attempted to create")
+                .register(Metrics.globalRegistry);
+        Counter topicsCreateFailed = Counter
+                .builder("numTopicsCreateFailed")
+                .description("Number of topics we've failed to create")
+                .register(Metrics.globalRegistry);
+        Counter topicsProduced = Counter
+                .builder("numTopicsProduced")
+                .description("Number of topics we've attempted to produce to")
+                .register(Metrics.globalRegistry);
+        Counter topicsProduceFailed = Counter
+                .builder("numTopicsProduceFailed")
+                .description("Number of topics we've failed to produce to")
+                .register(Metrics.globalRegistry);
+        Counter topicsConsumed = Counter
+                .builder("numTopicsConsumed")
+                .description("Number of topics we've attempted to consume from")
+                .register(Metrics.globalRegistry);
+        Counter topicsConsumeFailed = Counter
+                .builder("numTopicsConsumeFailed")
+                .description("Number of topics we've failed to consume from")
+                .register(Metrics.globalRegistry);
+
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> printMetrics(topicsCreated, topicsCreateFailed,
+                topicsProduced, topicsProduceFailed, topicsConsumed, topicsConsumeFailed,
+                firstMessageProduceTimeNanos, produceMessageTimeNanos, consumerReceiveTimeNanos,
+                consumerCommitTimeNanos)));
+
+        try (AdminClient kafkaAdminClient = KafkaAdminClient.create(kafkaAdminConfig);
+             KafkaProducer<Integer, byte[]> kafkaProducer = new KafkaProducer<>(kafkaProducerConfig)) {
+            ExecutorService createTopics = Executors.newFixedThreadPool(numConcurrentTopicCreations);
+            ExecutorService writeTopics = null;
+            if (numConcurrentProducers > 0) {
+                writeTopics = Executors.newFixedThreadPool(numConcurrentProducers);
+            }
+            ExecutorService consumeTopics = null;
+            if (numConcurrentConsumers > 0) {
+                consumeTopics = Executors.newFixedThreadPool(numConcurrentConsumers);
+            }
+            ExecutorService printMetrics = Executors.newSingleThreadExecutor();
+            printMetrics.submit((Runnable) () -> {
+                while (true) {
+                    try {
+                        TimeUnit.SECONDS.sleep(10);
```

Comment:
Nit: Do we want a constant for this for readability?

Reply:
Yup. Will move.